### PR TITLE
MAGECLOUD-3651: Update description of variables that support _merge

### PR DESCRIPTION
**Pull request by bbatsche**
### Description

Variables:

- QUEUE_CONFIGURATION
- SEARCH_CONFIGURATION
- ELASTICSUITE_CONFIGURATION
- CACHE_CONFIGURATION
- SESSION_CONFIGURATION
- DATABASE_CONFIGURATION
- RESOURCE_CONFIGURATION

Make the formatting consistent and make sure all include description of `_merge: true`

### Fixed Issues (if relevant)

[MAGECLOUD-3651](https://magento2.atlassian.net/browse/MAGECLOUD-3651)

### Manual testing scenarios

None, changes were in a static file

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: dist/.magento.env.yaml [hunk 1]
```diff
@@ -261,7 +266,12 @@
 #                  host: mq.host                                                                                      #
 #                  port: 1234                                                                                         #
 #######################################################################################################################
-# SEARCH_CONFIGURATION - use this variable to retain customized search service settings between deployments.          #
+# SEARCH_CONFIGURATION - Replace or modify the Magento search configuration generated by ece-tools during the Cloud   #
+#                        deployment process. The generated configuration is based on the available relationships in   #
+#                        the .magento.app.yaml file.                                                                  #
```
Comment (oshmyheliuk) [inline dist/.magento.env.yaml]: > The generated configuration is based on the available relationships in the .magento.app.yaml file.  

Is it true for pro environments? We have environments where services are adding by request, so we actually read relationships from the Environment variable and relationships in  .magento.app.yaml can be not valid.

File: dist/.magento.env.yaml [hunk 2]
```diff
@@ -336,9 +351,11 @@
 #                  page_cache:                                                                                        #
 #                    backend: file                                                                                    #
 #######################################################################################################################
-# SESSION_CONFIGURATION - use to configure Redis session storage. You must specify the save, redis, host, port,       #
-#                         and database options for the session storage variable.                                      #
-#                         By default, the deployment process overwrites all settings in the env.php file.             #
+# SESSION_CONFIGURATION - Replace or modify the Magento session configuration generated by ece-tools during Cloud     #
+#                         deployment. By default, ece-tools configures Magento to store session data in Redis.        #
```
Comment (oshmyheliuk) [inline dist/.magento.env.yaml]: > By default, ece-tools configures Magento to store session data in Redis.

This is true in cases when Redis service is configured

Comment (meker12) [inline dist/.magento.env.yaml]:
```suggestion
#                         deployment. By default, ece-tools configures Magento to store session data in Redis when the Redis service is configured.        #
```

File: dist/.magento.env.yaml (hunk 1 above)
Comment (bbatsche) [reply dist/.magento.env.yaml]: Bleh, you're right. I'll drop out the `.magento.app.yaml` portion; just "available relationships"

**Review by oshmyheliuk — DISMISSED** [review]

**Review by oshmyheliuk — DISMISSED** [review]

File: dist/.magento.env.yaml [hunk 3]
```diff
@@ -275,9 +284,12 @@
 #                elasticsearch_index_prefix: magento                                                                  #
 #                elasticsearch_server_timeout: '15'                                                                   #
 #######################################################################################################################
-# ELASTICSUITE_CONFIGURATION - use this variable to retain customized ElasticSuite service settings                   #
-# between deployments. All configs provided in this configuration are nested in main ElasticSuite                     #
-# config section 'system/default/smile_elasticsuite_core_base_settings'                                               #
+# ELASTICSUITE_CONFIGURATION - Customize ElasticSuite service configuration stored in the                             #
+#                              'system/default/smile_elasticsuite_core_base_settings' section of config.php.          #
```
Comment (meker12) [inline dist/.magento.env.yaml]:
```suggestion
#                              'system/default/smile_elasticsuite_core_base_settings' section of the config.php file.          #
```

File: dist/.magento.env.yaml [hunk 4]
```diff
@@ -349,9 +350,25 @@
 #                  page_cache:                                                                                        #
 #                    backend: file                                                                                    #
 #######################################################################################################################
-# SESSION_CONFIGURATION - use to configure Redis session storage. You must specify the save, redis, host, port,       #
-#                         and database options for the session storage variable.                                      #
-#                         By default, the deployment process overwrites all settings in the env.php file.             #
+# LOCK_PROVIDER - use to set the lock provider.                                                                       #
```
Comment (meker12) [inline dist/.magento.env.yaml]:
```suggestion
# LOCK_PROVIDER - Use to set the lock provider.                                                                       #
```

File: dist/.magento.env.yaml [hunk 5]
```diff
@@ -349,9 +350,25 @@
...
+# Default value: "file"                                                                                               #
+# Stages: deploy                                                                                                      #
+# Example:                                                                                                            #
+#          stage:                                                                                                     #
+#            deploy:                                                                                                  #
+#              LOCK_PROVIDER: "db"                                                                                    #
+#######################################################################################################################
+# SESSION_CONFIGURATION - Replace or modify the Magento session configuration generated by ece-tools during Cloud     #
+#                         deployment. By default, ece-tools configures Magento to store session data Redis, if it's   #
```
Comment (meker12) [inline dist/.magento.env.yaml]:
```suggestion
#                         deployment. By default, ece-tools configures Magento to store Redis session data.           #
```

File: dist/.magento.env.yaml [hunk 6]
```diff
@@ -349,9 +350,25 @@
...
+#                         available.                                                                                  #
```
Comment (meker12) [inline dist/.magento.env.yaml]:
```suggestion
```

File: dist/.magento.env.yaml [hunk 7]
```diff
@@ -371,10 +388,12 @@
...
-#                         By default, the deployment process overwrites all settings in the env.php file.             #
-#                         In case when added option "_merge: true" setting are merging with default configuration.    #
-#                                                                                                                     #
+# DATABASE_CONFIGURATION – Customize the Magento database connection information generated by ece-tools during the    #
+#                          deployment process.                                                                        #
+#                          To replace the existing configuration, specify values for each configuration option        #
+#                          required for your environment. To modify the existing configuration generated by           #
+#                          ece-tools, specify values only for the  options to add or update. Then, add the            #
+#                          `_merge: true` option.                                                                     #
```
Comment (meker12) [inline dist/.magento.env.yaml]: The second occurrence of  "generated by ece-tools" seems unnecessary.

`To modify the existing configuration, specify values only for the options to add or update. Then, add the `merge: true` option.`

File: dist/.magento.env.yaml [hunk 8]
```diff
@@ -409,15 +428,12 @@
...
-#                         By default, the deployment process overwrites all settings in the env.php file;             #
-#                         however, you can choose to merge one or more values for a service configuration without     #
-#                         overwriting all of the values.                                                              #
-#                                                                                                                     #
-#                         Set the _merge option to one of the following:                                              #
-#                         true - Merge the configured service values with the environment variable values.            #
-#                         false - Overwrite the configured service values with the environment variable values.       #
-#                                                                                                                     #
+# RESOURCE_CONFIGURATION - Replace or modify the data connection resource configuration generated during deployment.  #
```
Comment (meker12) [inline dist/.magento.env.yaml:426]: so here we say "generated during deployment".  That is shorter and clearer than the phrase  "generated by ece-tools during the deployment process".   Consider replacing the longer phrase with the shorter one.

File: dist/.magento.env.yaml [hunk 9]
```diff
@@ -409,15 +428,12 @@
...
+#                          This configuration maps a resource name to a database connection.                          #
+#                          To replace the existing configuration, specify values for each configuration option        #
+#                          needed for your environment. To modify the existing configuration generated by ece-tools   #
+#                          specify values only for the options to add or update. Then, add the `_merge: true`         #
```
Comment (meker12) [inline dist/.magento.env.yaml]: To modify the existing configuration generated by ece-tools ==> To modify the existing configuration, ...

**Review by meker12 — CHANGES_REQUESTED** [review]
A few small issues.

**Review by meker12 — APPROVED** [review]
LGTM +1 OK to merge.

**Review by oshmyheliuk — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Pull request merged**